_studio/mfx_lib/encode_hw/h264/src/mfx_h264_encode_hw.cpp

```diff
@@ -1133,6 +1133,21 @@ mfxStatus ImplementationAvc::Init(mfxVideoParam * par)
         MFX_CHECK_STS(sts);
     }
 
+    if (IsMctfSupported(m_video))
+    {
+        if (!m_cmDevice)
+        {
+            m_cmDevice.Reset(TryCreateCmDevicePtr(m_core));
+            if (!m_cmDevice)
+                return MFX_ERR_UNSUPPORTED;
+        }
+
+        amtMctf = std::make_shared<CMC>();
+        MFX_CHECK_NULL_PTR1(amtMctf);
```

Comment:
this is useless check, until you don't use ```nothrow``` in ```new```, ```make_shared``` will throw an exception, but not return ```nullptr```

Reply:
removed